src/main/java/net/minecraftforge/client/event/EntityViewRenderEvent.java

```diff
@@ -83,23 +90,74 @@ public void setDensity(float density)
     }
 
     /**
-     * Event that allows any feature to customize the rendering of fog.
+     * This event allows for customization of parameters related to fog rendering. The plane distances are based on the player's render distance.
+     * For the event to have an effect, you must cancel it.
+     *
+     * The FogMode is NOT customizable. It describes the type of fog being modified.
+     * A FogMode of FOG_SKY is the skybox's fog.
+     * A FogMode of FOG_TERRAIN is what the fog induced by render distance uses. This works best for reducing the camera's visibility.
      */
-    @HasResult
-    public static class RenderFogEvent extends FogEvent
+    @HasResult // TODO: remove in 1.19. Setting a result for this event has no effect.
+    @Cancelable
+    public static class RenderFogEvent extends FogEvent // TODO: In 1.19, change superclass to EntityViewRenderEvent
     {
-        private final float farPlaneDistance;
-
+        private final FogMode type;
+        private float farPlaneDistance;
+        private float nearPlaneDistance;
+        private FogShape fogShape;
+
+        /**
+         * @deprecated Use other constructor with all the params. Will be removed in 1.19
+         */
+        @Deprecated(forRemoval = true, since = "1.18.2")
         public RenderFogEvent(FogMode type, Camera camera, float partialTicks, float distance)
+        {
+            this(type, camera, partialTicks, -8f, distance, FogShape.SPHERE);
+        }
+
+        public RenderFogEvent(FogMode type, Camera camera, float partialTicks, float nearPlaneDistance, float farPlaneDistance, FogShape fogShape)
         {
             super(type, camera, partialTicks);
-            this.farPlaneDistance = distance;
+            this.type = type;
+            setFarPlaneDistance(farPlaneDistance);
+            setNearPlaneDistance(nearPlaneDistance);
+            setFogShape(fogShape);
+        }
+
+        public FogMode getMode()
+        {
+            return type;
         }
 
         public float getFarPlaneDistance()
         {
             return farPlaneDistance;
         }
+
+        public float getNearPlaneDistance()
+        {
+            return nearPlaneDistance;
+        }
+
+        public FogShape getFogShape()
+        {
+            return fogShape;
+        }
+
+        public void setFarPlaneDistance(float distance)
+        {
+            farPlaneDistance = distance;
+        }
+
+        public void setNearPlaneDistance(float distance)
+        {
+            nearPlaneDistance = distance;
+        }
+
+        public void setFogShape(FogShape shape)
+        {
+            fogShape = shape;
+        }
```

Comment:
You could make incrementing setters as well for these to make it such that adding onto the existing value is not simply `set(get + newVal)`.